x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -38,4 +136,120 @@ public IngestDocument execute(IngestDocument ingestDocument) {
+            this.ingestService = ingestService;
+            clusterService.getClusterSettings().addSettingsUpdateConsumer(MAX_INFERENCE_PROCESSORS, this::setMaxIngestProcessors);
+        }
+
+        @Override
+        public void accept(ClusterState state) {
+            minNodeVersion = state.nodes().getMinNodeVersion();
+            MetaData metaData = state.getMetaData();
+            if (metaData == null) {
+                currentInferenceProcessors = 0;
+                return;
+            }
+            IngestMetadata ingestMetadata = metaData.custom(IngestMetadata.TYPE);
+            if (ingestMetadata == null) {
+                currentInferenceProcessors = 0;
+                return;
+            }
+
+            int count = 0;
+            for (PipelineConfiguration configuration : ingestMetadata.getPipelines().values()) {
+                try {
+                    Pipeline pipeline = Pipeline.create(configuration.getId(),
+                        configuration.getConfigAsMap(),
+                        ingestService.getProcessorFactories(),
+                        ingestService.getScriptService());
+                    count += pipeline.getProcessors().stream().filter(processor -> processor instanceof InferenceProcessor).count();
+                } catch (Exception ex) {
+                    logger.warn(new ParameterizedMessage("failure parsing pipeline config [{}]", configuration.getId()), ex);
+                }
+            }
+            currentInferenceProcessors = count;
+        }
+
+        // Used for testing
+        int numInferenceProcessors() {
+            return currentInferenceProcessors;
+        }
+
+        @Override
+        public InferenceProcessor create(Map<String, Processor.Factory> processorFactories, String tag, Map<String, Object> config)
+            throws Exception {
+
+            if (this.maxIngestProcessors <= currentInferenceProcessors) {
+                throw new ElasticsearchStatusException("Max number of inference processors reached, total inference processors [{}]. " +
+                    "Adjust the setting [{}]: [{}] if a greater number is desired.",
+                    RestStatus.CONFLICT,
+                    currentInferenceProcessors,
+                    MAX_INFERENCE_PROCESSORS.getKey(),
+                    maxIngestProcessors);
+            }
+
+            String modelId = ConfigurationUtils.readStringProperty(TYPE, tag, config, MODEL_ID);
+            String targetField = ConfigurationUtils.readStringProperty(TYPE, tag, config, TARGET_FIELD);
+            Map<String, String> fieldMapping = ConfigurationUtils.readOptionalMap(TYPE, tag, config, FIELD_MAPPINGS);
+            InferenceConfig inferenceConfig = inferenceConfigFromMap(ConfigurationUtils.readMap(TYPE, tag, config, INFERENCE_CONFIG));
+            String modelInfoField = ConfigurationUtils.readStringProperty(TYPE, tag, config, MODEL_INFO_FIELD, "_model_info");
```

Comment:
By default, I think we should add a field that includes the model ID used in the inference step. 

This should probably append the `tag` to the end of the default info field to protect against multiple processors in the same pipeline...